optimizely/event/user_event_factory.py

```diff
@@ -49,11 +49,12 @@ def create_impression_event(
         if activated_experiment:
             experiment_id = activated_experiment.id
 
-        if variation_id and experiment_id:
-            variation = project_config.get_variation_from_id_by_experiment_id(experiment_id, variation_id)
-        # need this condition when we send events involving forced decisions
-        elif variation_id and flag_key:
+        if variation_id and flag_key:
             variation = project_config.get_flag_variation(flag_key, 'id', variation_id)
+        if not variation and variation_id and experiment_id:
```

Comment:
```suggestion
        elif variation_id and experiment_id:
```
We need to keep this to support legacy `activate` impression event.